meetings/2022-03-21.md

```diff
@@ -0,0 +1,137 @@
+
+## Minutes
+
+### FPWD Deadline
+
+* AC: I think we should set specific data
+* eP: We are pretty much ready, will take a last pass and check for in-line issues
+* AC: We've been talkin about conformance classes. Should we wait for it.
+* eP: good question (started with small pieces, script tags, etc). It would be better not to delay. We are clear that there are a few things to do. Will make a simple PR for this next week
+* AC: How about a next week deadline? (Week of March 28, 2022)
+* eP: +1
+* eP: conformance classes can be added later, if they don't make it into the FPWD
+* eP: there was an issue about namespaces
+
+ACTION: AC will write PR to address [namespaces issue](https://github.com/solid/solid-oidc/issues/23)
+ACTION: eP to add any missing issues inline
+ACITON: eP to add first version of conformance clases
+
+### Specify the client authentication method when using a Client Identifier
+
+[Issue#78](https://github.com/solid/solid-oidc/issues/78)
+
+* eP: https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication
+
+> none
+> The Client does not authenticate itself at the Token Endpoint, either because it uses only the Implicit Flow (and so does not use the Token Endpoint) or because it is a Public Client with no Client Secret or other authentication mechanism.
+
+* eP: we should clarify what confidential clients do
+* AC: are we going to restate what OIDC or OAuth2 already says about it?
+* eP: How do we see this working in practice? In some implementations, it seems to work with `none`?
+* AvdB: the spec mentions client registration. Where would the `none` be set?
+* AvdB: when you register a client (e.g., Dynamic Client Registration), you can set a particular method. For Client Identifiers, these are public clients
+* eP: I wasn't sure if this is for the token endpoint
+* AC: Nature of client identfires is that these are public clients. If it is confidential clients, the authentication method is important. Those client identifiers are intended to be public clients. Maybe we should be clear about it the spec.
+* eP: for use with confidential clients, in interop panel, the authorization agent acts as a confidential client
+* AC: what we define is about Public Clients but we don't say anything about confidential clients. If impl supports confidential clients it can do anything it likes provided that it follows OAuth 2.0.
+* eP: can Public Client get a refresh token?
+* AC: There needs to be interaction with end-user to confirm refresh token.
+* AC: I'm not a fan of refresh tokens in the browser, but given browser limitations around iframes, they are useful in certain cases.
+* eP: it's recommended that an ID token is short-lived, so how do you get a token that isn't expired
+* AC: One approach is using refresh tokens, another approach is rather than redirecting user away, open new window and direct user through the auth flow. Given that the have cookie there should not be any interaction
```

Comment:
```suggestion
* eP: it's recommended that an ID token is short-lived, so how do you get a token that isn't expired?
* AC: One approach is using refresh tokens. Another approach is, rather than redirecting user away, open new window, and direct user through the auth flow. Given that they have cookie, there should not be any interaction.
```